platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompilationFailedException.java

```diff
@@ -27,8 +29,9 @@
 import java.util.List;
 import java.util.Optional;
 
-public class CompilationFailedException extends RuntimeException implements CompilationFailedIndicator, ProblemAwareFailure {
+public class CompilationFailedException extends RuntimeException implements CompilationFailedIndicator, ProblemAwareFailure, ResolutionProvider {
 
+    public static final String RESOLUTION_MESSAGE = "Check you code and dependencies to fix the compilation error(s)";
```

Comment:
```suggestion
    public static final String RESOLUTION_MESSAGE = "Check your code and dependencies to fix the compilation error(s)";
```

Reply:
Can we add a test to verify this message is printed?

Reply:
We have a test:
https://github.com/gradle/gradle/pull/30853/files#diff-fa5145b319124236d7807fd36a6af74ccebf240372d4a22a6413c3e1bde2e983